src/plugins/share/common/url_service/locators/types.ts

```diff
@@ -50,7 +50,7 @@ export interface LocatorDefinition<P extends SerializableState>
 /**
  * Public interface of a registered locator.
  */
-export interface LocatorPublic<P> {
+export interface LocatorPublic<P extends SerializableState> extends PersistableState<P> {
```

Comment:
this should  be PersistableStateDefinition to make it optional to provide certain things like migrations

Reply:
I think it should be `PersistableState` as it is now, it is good to have all properties of `PersistableState` be implemented on `LocatorPublic`. Maybe you mean `LocatorDefinition`? In `LocatorDefinition` it uses partial of `PersistableState`, exactly to make properties optional.

Reply:
why would a locator need to implement inject and extract if state does not contain any saved object references ?

Reply:
It does not have to. When you define a locator you would just skip the `inject` and `extract` methods and on this `LocatorPublic` interface they would be correctly implemented to the default no-op functions.